app/models/commit.rb

```diff
@@ -173,6 +173,12 @@ def last_statuses
     @last_statuses ||= statuses.to_a.uniq(&:context).sort_by(&:context).presence || [UnknownStatus.new(self)]
   end
 
+  def status
+    visibles = visible_statuses
+    status = visibles.size > 1 ? StatusGroup.new(significant_status, visibles) : visibles.first
+    missing_statuses.empty? ? status : MissingStatus.new(status, missing_statuses)
+  end
```

Comment:
Then no. Because if you have a single status you don't go through the group path.